### PR TITLE
Meta-RFC - RFC about the updating the RFC process

### LINKED ISSUE #4738: RFC Process Improvements
The RFC documents in our project have proven themselves to be a successful way to communicate designs for larger projects. When feedback is given, we're able to adjust and iterate.

However, there has been some friction.

Friction Points:
* Marking RFCs as "reviewed" in one way or another from relevant stakeholders takes too long
* Criteria for approving RFCs is outdated -- Originally it was set up in such a way that newer folks to the team would have someone who has more context to verify that all the context was considered. This needs to be revisited.
* Criteria for landing RFCs has always been nebulous -- Should it land when it's approved? When it's implemented?
* CI runs on RFC and documentation changes
* TODO

Action Items:
- [ ] Propose a new process - Aneesha
	(use "request changes" / "approve" buttons)
- [ ] Explore technical solutions to prevent friction (bots? ci fixes?)

Exit Criteria:
1. Change required reviewers to a relevant set of folks (tech leads?)
2. CI shouldn't run unrelated jobs (only markdown linting or something)
3. Institute culture of new process via shared in eng group (async or meeting)

### PR DESCRIPTION
**Pull request by aneesharaines**
This is a RFC that describes suggested changes to the current RFC process with the goal of reducing the amount of time RFCs are open without action.  

Goal is to close this RFC by July 31st.  

Closes #4738

### CODE REVIEW

File: rfcs/taskforce/0004-rfc-process-update.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+**Owner/Champion**
+
+Regardless if it's one person or a focused group of people who are generating the RFC, there should be an owner or champion who pushes the RFC forward.  In most cases, this is be the person writing the RFC.
+
+**The RFC Iteration Process**
+
+- Before submitting an RFC it is generally a good idea to pursue feedback from other project developers beforehand, to ascertain that the RFC and approach is desirable; having a consistent impact on the project requires concerted effort toward consensus building.
+- To make an RFC, just copy the [0000-template.md](http://0000-template.md/) to [0000-shortname.md](http://0000-shortname.md/). It's also a good idea to submit drafts of RFCs as PRs for greater visibility.
+- When you are ready to submit your RFC, create a PR if it hasn't already been done (make sure to remove the [DRAFT] in the title) and request a reviews from the appropriate team or team members.
```
Comment (bkase) [inline rfcs/taskforce/0004-rfc-process-update.md:40]: we can use the "draft PR" feature of github for this too

File: rfcs/taskforce/0004-rfc-process-update.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+- Discuss and incorporate feedback - This is an active process, with the goal to resolve and RFC withing a week after submitting for review, unless substantial changes are required.  The owner / champion should drive this process, making sure to follow up with the team members whose approval is required to get the necessary reviews.  To facilitate this process:
```
Comment (bkase) [inline rfcs/taskforce/0004-rfc-process-update.md:41]: How does one determine whose approval is required? Is it up to the driver of the RFC? Are there expectations around how many reviewers are needed? Is this something to bring up in a team meeting?

File: rfcs/taskforce/0004-rfc-process-update.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+- Addressing open or unresolved questions
+- Revisiting and updating the approach if it has been changed between the RFC being merged and the
+
+RFCs should not be updated to reflect actual implementation. That should be captured in docs and done as part of the development process (as captured in the Definition of Done - TBA)
+
+**The RFC Template**
+
+Suggest adding the following section(s) to the RFC Template:
+
+- Testing - discussing how this feature or changes should be tested and if there's anything (eg. infrastructure) that needs to be done to support that
```
Comment (bkase) [inline rfcs/taskforce/0004-rfc-process-update.md:72]: 👍

File: rfcs/taskforce/0004-rfc-process-update.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+
+## Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Should we RFC substantial changes to our processes as well? Might be good for visibility.
+- Should we have a code owners team for the rfc folder. Leaning towards no but worth discussing
```
Comment (bkase) [inline rfcs/taskforce/0004-rfc-process-update.md:79]: We used to and this slowed us down a lot, but see my comment above -- we should set clear expectations around who and how many approvals are needed.

File: rfcs/taskforce/0004-rfc-process-update.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+[summary]: #summary
+
+The RFC (Request For Comments) is a process by which we share technical designs and gain consensus on how to proceed. It is intended to provide a consistent and controlled path for new features to enter the project. While the RFCs can be made by anyone, including community members, RFCs that don’t come from the core engineering team will need a champion to drive the RFC through the consensus process. 
+
+## Motivation
+[motivation]: #motivation
+
+What problems are we tring to address?
+
+- Long Lived Open RFC Prs - We don’t have a good process for closing out RFCs, as a consequence, there are quite a few open RFCs (19) the majority of which are not active. It’s also pretty co
```
[the remark this answers is not in the document]
Comment (aneesharaines) [reply rfcs/taskforce/0004-rfc-process-update.md:11]: True!  will fix that.

**Pull request closed without merging**